benchmark_v2/gather_benchmark.py

```diff
@@ -0,0 +1,97 @@
+import arkouda as ak
+import pytest
+import numpy as np
+
+TYPES = ("int64", "float64", "bool", "str")
+
+
+def _ak_gather(a, i):
+    """
+    Helper function allowing for the gather to be benchmarked
+    """
+    return a[i]
+
+
+@pytest.mark.benchmark(group="AK_Gather")
+@pytest.mark.parametrize("dtype", TYPES)
+def bench_ak_gather(benchmark, dtype):
+    cfg = ak.get_config()
+    isize = pytest.prob_size if pytest.idx_size is None else pytest.idx_size
+    vsize = pytest.prob_size if pytest.val_size is None else pytest.val_size
+    Ni = isize * cfg["numLocales"]
+    Nv = vsize * cfg["numLocales"]
+
+    if dtype in pytest.dtype:
+        i = ak.randint(0, Nv, Ni, seed=pytest.seed)
+        if pytest.seed is not None:
+            pytest.seed += 1
+        if pytest.random or pytest.seed is not None:
+            if dtype == "int64":
+                v = ak.randint(0, 2 ** 32, Nv, seed=pytest.seed)
+            elif dtype == "float64":
+                v = ak.randint(0, 1, Nv, dtype=ak.float64, seed=pytest.seed)
+            elif dtype == "bool":
+                v = ak.randint(0, 1, Nv, dtype=ak.bool, seed=pytest.seed)
+            elif dtype == "str":
+                v = ak.random_strings_uniform(1, 16, Nv, seed=pytest.seed)
+        else:
+            if dtype == "str":
+                v = ak.cast(ak.arange(Nv), "str")
+            else:
+                v = ak.ones(Nv, dtype=dtype)
+
+        c = benchmark.pedantic(_ak_gather, args=[v, i], rounds=pytest.trials)
+
+        if dtype == "str":
+            offsets_transferred = 3 * c.size * 8
+            bytes_transferred = (c.size * 8) + (2 * c.nbytes)
+            bytes_per_sec = (offsets_transferred + bytes_transferred) / benchmark.stats["mean"]
+        else:
+            bytes_per_sec = (c.size * c.itemsize * 3) / benchmark.stats["mean"]
+
+        bytes_per_sec / 2 ** 30
```

Comment:
I don't think we need this here since we're doing the same operation in the transfer rate comment and this isn't being used